ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Cleaner.java

```diff
@@ -179,6 +180,13 @@ private void clean(CompactionInfo ci, long minOpenTxnGLB, boolean metricsEnabled
         txnHandler.markCleaned(ci);
         return;
       }
+      if (MetaStoreUtils.isNoCleanUpSet(t.getParameters())) {
+        // The table was marked no clean up true.
+        LOG.info("Skipping " + ci.getFullTableName() + " clean up, as NO_CLEANUP set to true");
+        txnHandler.markCleaned(ci);
```

Comment:
if we won't call ````markCleaned```` that would lead to the accumulation of COMPACTION_QUEUE entries in READY_FOR_CLEANING state (mil of duplicates) + infinite re-try every 5 sec (default). 
However, if we do - some of the obsolete files could stay forever. 
Would it be sufficient to have a config that disables the Cleaner completely?

Reply:
@deniskuzZ I agree with you there will be some obsolete files could stay forever only if user is replying on auto compaction. This config is for user who manually trigger compaction in there data pipelines. i.e 

ALTER TABLE table_name COMPACT 'major' WITH OVERWRITE TBLPROPERTIES ("no_cleanup"="true"); 

{some ETL operation}

ALTER TABLE table_name COMPACT 'major' WITH OVERWRITE TBLPROPERTIES ("no_cleanup"="false"); 

Any hive query/ETL pipeline always refer to handful of table. So disabling Cleaner completely will increase the blast radius and over head for cleaner. if a user forget to revert no_cleanup back to false then it will have performance impact which we should document in hive wiki.

Reply:
@deniskuzZ Yes, you have a valid point. I think, we can just document it.